### PR TITLE
try to fix sorting by duration in Computer feature (lp1916560)

### PR DESCRIPTION
**Pull request by Swiftb0y**
This is my attempts at fixing [lp1916560](https://bugs.launchpad.net/mixxx/+bug/1916560). I'm not familiar with the code so I'm not confident this won't introduce any regressions, but it successfully mitigates the issue described in the bug report.

### CODE REVIEW

**Comment by Be-ing** [comment]
Should this target the 2.3 branch?

**Comment by ronso0** [comment]
Please describe the issue in the PR title so it can be found later on we looking through the PR list, like "fix sorting by duration in Computer feature" or something like that

**Review by uklotzde — APPROVED** [review]
Seems to be reasonable, didn't test.

File: src/library/browse/browsethread.cpp [hunk 1]
```diff
@@ -211,7 +211,7 @@ void BrowseThread::populateModel() {
             QString duration = pTrack->getDurationText(mixxx::Duration::Precision::SECONDS);
             item = new QStandardItem(duration);
             item->setToolTip(item->text());
-            item->setData(item->text(), Qt::UserRole);
+            item->setData(pTrack->getDurationInt(), Qt::UserRole);
```
Comment (uklotzde) [inline src/library/browse/browsethread.cpp:214]: Does `double getDuration()` also work? No need to round to integers here.

Comment (Swiftb0y) [reply src/library/browse/browsethread.cpp:214]: Oh, of course, yeah that also works.

**Review by uklotzde — APPROVED** [review]
I didn't test it myself yet, but this should fix the sorting issues. Thanks for the quick fix! LGTM

Please check if your are able to mark the bug as "Fix committed" on Launchpad.

**Comment by uklotzde** [comment]
@Swiftb0y Please merge *2.3* into *main* and then push upstream ;)

**Comment by Be-ing** [comment]
It would be great if we automate these merges and send a notification to Zulip if there is a conflict.

**Comment by Holzhaus** [comment]
> It would be great if we automate these merges and send a notification to Zulip if there is a conflict.

Note that there might be conflict even if git does not complain. So if we automate, then it needs to build the merge and run the tests before anything is pushed.

**Comment by Swiftb0y** [comment]
> Please merge 2.3 into main and then push upstream ;)

Can you link me a guide or something? I know git quite a bit, but not enough to feel comfortable pushing directly upstream (without someone else confirming I didn't mess up somewhere).

**Comment by Holzhaus** [comment]
> > Please merge 2.3 into main and then push upstream ;)
> 
> Can you link me a guide or something? I know git quite a bit, but not enough to feel comfortable pushing directly upstream (without someone else confirming I didn't mess up somewhere). 

```
$ git checkout main
$ git fetch upstream && git reset --hard upstream/main
```
(just to make sure we use the correct main branch) 

Then:
```
$ git pull upstream 2.3
```

Now build mixxx and run the tests. 

If everything works fine:

```
git push upstream main
```

**Comment by Swiftb0y** [comment]
Thanks for the guide. my default commit message seems to be different (` Merge branch '2.3' of https://github.com/mixxxdj/mixxx
`) from what others usually have (`Merge remote-tracking branch 'upstream/2.3' into main`), but I inspected the commit and it seemed fine.

**Comment by Holzhaus** [comment]
I have the same commit message as you. I suppose the other team members use a different `git` version or config.

**Comment by Be-ing** [comment]
My remote for mixxxdj/mixxx is called `upstream`. If yours is `origin` that may explain why it isn't automatically in the merge commit messages.

**Comment by Holzhaus** [comment]
The difference comes from the git command.

1. `git pull upstream 2.3` -> `Merge branch '2.3' of https://github.com/mixxxdj/mixxx`
2. `git merge upstream/2.3` -> `Merge remote-tracking branch 'upstream/2.3' into main`

I always use the first one.

**Comment by uklotzde** [comment]
I mostly use `git gui` and always merge the tracking `upstream` branch, not the local branch even if both are in sync. The result just differs by the generated commit message.

**Comment by uklotzde** [comment]
This corresponds to the command `git pull upstream 2.3`.

**Comment by Swiftb0y** [comment]
Thanks for clarifying. :)

**Pull request merged**